students/srepking/Lesson04/mailroom2.py

```diff
@@ -0,0 +1,122 @@
+            if i != 0:
+                count = count + 1
+        # Calculate total donation amount
+        if count > 0:
+            average = total / count
+        donors_f[name] = [total, count, average]
+    print('\n')
+    # Find the longest name in the list to format column width
+    name_list = list(donors_f.keys())  # creates a list of keys
+    name_wi = 11
+    for i in name_list:
+        if len(i) > name_wi:
+            name_wi = (len(i))  # width of name column
+    # Find the longest donation amount
+    tot_wi = 12
+    ave_wi = 12
+    num_wi = 12
+    for name, summary in donors_f.items():
+        if len(str(summary[0])) > tot_wi:
+            # width of total column
+            tot_wi = (len(str(summary[0]))) + 3
+            # width of number of donations column
+        if len(str(summary[1])) > num_wi:
+            num_wi = (len(str(summary[1]))) + 3
+            # width of total average column
+        if len(str(int(summary[2]))) > ave_wi:
+            ave_wi = (len(str(int(summary[2])))) + 3
+    list_sorted = sorted(donors_f, key=donors_f.__getitem__, reverse=True)
+    # Print the Table
+    print(f"{'Donor Name':{name_wi}}| {'Total Given':^{tot_wi}}| "
+          f"{'Num Gifts':^{num_wi}}| {'Average Gift':^{ave_wi}}")
+    print(f"{'-':-^{(name_wi+tot_wi+ave_wi+num_wi+8)}}")
+    # print("-")
+    for key in list_sorted:
+        temp = donors_f[key]
+        print(f"{key:{name_wi}}${temp[0]:{tot_wi}}{temp[1]:^{num_wi}}   "
+              f"${temp[2]:>{ave_wi}.2f}")
+
+    print('\n')
+    return donors_f
+
+
+def letters_for_all():
+    path_letters = os.getcwd()
+    print(f"You chose to send letters for everyone. "
+          f"The letters have been completed and you "
+          f"can find them here: {path_letters}")
+    donors_f = report()
+    for donor, donation in donors.items():
+        donation_summary = donors_f[donor]
+        letter = f'Dear {donor}, thank you so much for your last ' \
+                 f'contribution of ${donation[-1]:.2f}! ' \
+                 f'You have contributed a total of ${donation_summary[0]:.2f}, ' \
+                 f'and we appreciate your support!'
+        # Write the letter to a destination
+        with open(donor + '.txt', 'w') as to_file:
+            to_file.write(letter)
+
+
+def goodbye():
+    pass
+
+
+if __name__ == '__main__':
+    import os
+
+    donors = {'Joe Edgar Allen Poe the Third ': [1, 4, 200],
+              'Jack': [4, 5], 'Jill': [4], 'Jake': [.30],
+              'Jim': [1, 2, 1.04]}
+    while True:
+        response = input(
+            '\nChoose an Action:\n'
+            '1 - Send a Thank You\n'
+            '2 - Create a Report\n'
+
+            '3 - Send letters to everyone\n'
+            '4 - Quit\n'
+            '>>')
+        switch_dict = {'1': thank_you, '2': report, '3': letters_for_all}
+        switch_dict.get(response, goodbye)()
+        if response == '4':
+            print('Goodbye!')
+            break
```

Comment:
you can avoid special logic if you map option `4` to a `quit` or `exit` type function.